build.sbt

```diff
@@ -1,84 +1,138 @@
+    "Typesafe Releases" at "http://repo.typesafe.com/typesafe/releases/"
+  ),
 
-// stricter Scala compiler options
-scalacOptions in ThisBuild := Seq(
-  "-Xfatal-warnings",
-  "-feature",
-  "-deprecation:false",
-  "-unchecked",
-  "-Xfuture"
-)
+  // Add stricter Scala compiler options
+  scalacOptions in ThisBuild ++= Seq(
+    "-Xfatal-warnings",
+    "-feature",
+    "-deprecation:false",
+    "-unchecked",
+    "-Xfuture"
+  ),
 
-lazy val cleanup = settingKey[Boolean]("Set to false to disable cleaning up after each test")
-cleanup := true
-addCommandAlias("cleanupOn", "set cleanup := true")
-addCommandAlias("cleanupOff", "set cleanup := false")
+  // Common dependencies
+  libraryDependencies ++= Seq(
+    compilerPlugin("org.psywerx.hairyfotr" %% "linter" % "0.1.12"),
+    "com.typesafe.play" %% "play-json" % playVersion,  // TODO: Allow using other Json libraries
+    "org.scalacheck" %% "scalacheck" % scalacheckVersion,
+    "org.scalactic" %% "scalactic" % scalatestVersion,
+    "org.scalatest" %% "scalatest" % scalatestVersion % "test"
+  ),
 
-// call setup and teardown hooks before and after any tests run
-testOptions ++= Seq(
-  Tests.Setup {
-    classLoader: ClassLoader =>
-      val cleanDBAfterTests = cleanup.value
-      sys.props += "NEO4J_TEST_CLEANUP_AFTER" -> cleanDBAfterTests.toString
-      if (cleanDBAfterTests) {
-        println(AnsiColor.GREEN +
-          """All tests will cleanup their changes to their namespace in the database after they finish.
-            |Note: The tests will cleanup their own namespace from previous test runs before they run.""".stripMargin
-          + AnsiColor.RESET)
-      }
-      else {
-        println(AnsiColor.BLUE +
-          """Tests will NOT cleanup their changes to their namespace in the database after they finish.
-            |Note: The tests will cleanup their own namespace from previous test runs before they run.""".stripMargin
-          + AnsiColor.RESET)
-      }
-      classLoader.loadClass("me.jeffmay.neo4j.client.SetupBeforeTests").newInstance()
-  },
-  Tests.Cleanup {
-    classLoader: ClassLoader =>
-      classLoader.loadClass("me.jeffmay.neo4j.client.CleanupAfterTests").newInstance()
-  }
-)
+  // Avoid sbt warnings for these libraries
+  dependencyOverrides := Set(
+    "com.typesafe.play" %% "play-json" % playVersion,
+    "com.google.guava" % "guava" % "18.0",
+    "org.scalactic" %% "scalactic" % scalatestVersion
+  ),
+
+  // Don't publish tests.jar
+  publishArtifact in Test := false,
+  // Don't compile or publish ScalaDoc
+  // (ScalaDoc compiler breaks on links and the source code comments are more valuable)
+  sources in(Compile, doc) := Seq.empty,
+  publishArtifact in (Compile, packageDoc) := false
+
+) ++
+  // Publish to https://bintray.com/jeffmay/maven
+  bintraySettings ++ bintrayPublishSettings
+
+lazy val core = (project in file("core")).settings(commonSettings ++ Seq(
+  name := "neo4j-scala-models"
+))
```

Comment:
Core API and models. This also include generators, since I don't really know of a good way to put the tests in the same module and publish the generators. @chrissalij-r has been struggling with this for a while. I personally don't think having ScalaCheck as a source code dependency is all that bad. It is probably not going to cause evictions and it is a very small library.